src/rp2_common/pico_stdio_usb/include/pico/stdio_usb.h

```diff
@@ -94,10 +104,29 @@ extern "C" {
 extern stdio_driver_t stdio_usb;
 
 /*! \brief Explicitly initialize USB stdio and add it to the current set of stdin drivers
- *  \ingroup pico_stdio_uart
+ *  \ingroup pico_stdio_usb
+ *
+ *  \ref PICO_STDIO_USB_POST_CONNECT_WAIT_DELAY_MS can be set to cause this method to wait for a CDC connection
+ *  from the host before returning, which is useful if you don't want any initial stdout output to be discarded
+ *  before the connections is established.
+ *
+ *  \return true if the USB CDC was initialized, false if an error occurred
  */
 bool stdio_usb_init(void);
 
+/*! \brief Check if there is an active stdio CDC connection to a host
+ *  \ingroup pico_stdio_usb
+ *
+ *  This method can be used in a loop to wait for a connection to be established with the host
+ *  before starting to write to stdout (otherwise output is discarded).
+ *
+ *  Note that some host side terminal emulators seem to occasionally miss the first output
+ *  even though the connection is established, so you may also want to sleep a few milliseconds after this
+ *  method returns true.
+ *
+ *  \return true if stdio is connected over CDC
+ */
+bool stdio_usb_connected(void);
```

Comment:
I guess a common usage-idiom (for example-code that wants to be switchable between UART and USB-CDC using just the `CMakeLists.txt` mechanism) would be to wrap this inside a `#if LIB_PICO_STDIO_USB` ? Would that code need to also explicitly include `stdio_usb.h` inside a `#if LIB_PICO_STDIO_USB` too?
Or have I grabbed the totally wrong end of the stick? :upside_down_face:

Reply:
yes you would do exact all that... however the other #define stuff avoids you having to mess with stuff. I did not want to try to shoe-horn connection waiting into the abstraction for stdio drivers atm especially since the UART one is not capable of waiting even though you could detect a UART connection with more pins